File: scripts/05_exploratory_analysis.R
# %% Exploratory Data Analysis
# Script: 05_exploratory_analysis.R
# Purpose: Summary statistics and initial data exploration

# Load required libraries
library(dplyr)
library(tidyr)
library(ggplot2)

# Clear workspace
rm(list = ls())

# Set working directory
if (!dir.exists("scripts")) {
  setwd("..")
}

# Create output directories
dir.create("plots/exploratory", showWarnings = FALSE, recursive = TRUE)
dir.create("results/exploratory", showWarnings = FALSE, recursive = TRUE)

# Load data with exclusion flags
cat("Loading data...\n")
merged_fixation_data <- read.csv("data/processed/merged_fixation_data_with_flags.csv", stringsAsFactors = FALSE)
merged_interest_area_data <- read.csv("data/processed/merged_interest_area_data.csv", stringsAsFactors = FALSE)
merged_saccade_data <- read.csv("data/processed/merged_saccade_data_with_flags.csv", stringsAsFactors = FALSE)
merged_trial_data <- read.csv("data/processed/merged_trial_data_with_flags.csv", stringsAsFactors = FALSE)

# Filter out excluded data
fixation_clean <- merged_fixation_data %>% filter(!exclude_fixation)
saccade_clean <- merged_saccade_data %>% filter(!exclude_saccade)
trial_clean <- merged_trial_data %>% filter(!exclude_trial)

# %% Summary Statistics by Voice Type
cat("Calculating summary statistics by voice type...\n")

# Note: voice_type may need to be extracted from voice_id or other variables
# For now, we'll use voice_id as a proxy if voice_type doesn't exist
if (!"voice_type" %in% names(trial_clean)) {
  # Create a placeholder - this should be adjusted based on actual data
  trial_clean$voice_type <- "unknown"
  fixation_clean$voice_type <- "unknown"
  saccade_clean$voice_type <- "unknown"
}

# Fixation duration summary by voice type and AOI
fixation_summary <- fixation_clean %>%
  group_by(voice_type, aoi_type) %>%
  summarise(
    n = n(),
    mean_duration = mean(CURRENT_FIX_DURATION, na.rm = TRUE),
    median_duration = median(CURRENT_FIX_DURATION, na.rm = TRUE),
    sd_duration = sd(CURRENT_FIX_DURATION, na.rm = TRUE),
    min_duration = min(CURRENT_FIX_DURATION, na.rm = TRUE),
    max_duration = max(CURRENT_FIX_DURATION, na.rm = TRUE),
    .groups = "drop"
  )

write.csv(fixation_summary, "results/exploratory/fixation_summary_by_voice_aoi.csv", row.names = FALSE)

# Dwell time summary by voice type and AOI
dwell_time_summary <- merged_interest_area_data %>%
  filter(IA_LABEL != "" & !is.na(IA_LABEL)) %>%
  group_by(voice_id, aoi_type) %>%
  summarise(
    n = n(),
    mean_dwell_time = mean(IA_DWELL_TIME, na.rm = TRUE),
    median_dwell_time = median(IA_DWELL_TIME, na.rm = TRUE),
    sd_dwell_time = sd(IA_DWELL_TIME, na.rm = TRUE),
    .groups = "drop"
  )

write.csv(dwell_time_summary, "results/exploratory/dwell_time_summary_by_voice_aoi.csv", row.names = FALSE)

# Saccade summary by voice type
saccade_summary <- saccade_clean %>%
  group_by(voice_type) %>%
  summarise(
    n = n(),
    mean_amplitude = mean(CURRENT_SAC_AMPLITUDE, na.rm = TRUE),
    mean_duration = mean(CURRENT_SAC_DURATION, na.rm = TRUE),
    mean_velocity = mean(CURRENT_SAC_AVG_VELOCITY, na.rm = TRUE),
    .groups = "drop"
  )

write.csv(saccade_summary, "results/exploratory/saccade_summary_by_voice.csv", row.names = FALSE)

# %% Distribution plots for key dependent variables
cat("Creating distribution plots...\n")

# Fixation duration distribution
p1 <- ggplot(fixation_clean, aes(x = CURRENT_FIX_DURATION)) +
  geom_histogram(bins = 50, fill = "skyblue", color = "black", alpha = 0.6) +
  labs(title = "Distribution of Fixation Duration",
       x = "Fixation Duration (ms)",
       y = "Frequency") +
  theme_minimal()

ggsave("plots/exploratory/fixation_duration_distribution.png", p1, width = 8, height = 5)

# Fixation duration by voice type and AOI
p2 <- ggplot(fixation_clean %>% filter(!is.na(aoi_type)), 
             aes(x = voice_type, y = CURRENT_FIX_DURATION, fill = aoi_type)) +
  geom_boxplot(position = position_dodge(width = 0.8)) +
  labs(title = "Fixation Duration by Voice Type and AOI",
       x = "Voice Type",
       y = "Fixation Duration (ms)",
       fill = "AOI Type") +
  theme_minimal()

ggsave("plots/exploratory/fixation_duration_by_voice_aoi.png", p2, width = 10, height = 6)

# Dwell time distribution
p3 <- ggplot(merged_interest_area_data %>% filter(!is.na(IA_DWELL_TIME)), 
             aes(x = IA_DWELL_TIME)) +
  geom_histogram(bins = 50, fill = "lightgreen", color = "black", alpha = 0.6) +
  labs(title = "Distribution of Dwell Time",
       x = "Dwell Time (ms)",
       y = "Frequency") +
  theme_minimal()

ggsave("plots/exploratory/dwell_time_distribution.png", p3, width = 8, height = 5)

# Saccade amplitude distribution
p4 <- ggplot(saccade_clean, aes(x = CURRENT_SAC_AMPLITUDE)) +
  geom_histogram(bins = 50, fill = "salmon", color = "black", alpha = 0.6) +
  labs(title = "Distribution of Saccade Amplitude",
       x = "Saccade Amplitude (degrees)",
       y = "Frequency") +
  theme_minimal()

ggsave("plots/exploratory/saccade_amplitude_distribution.png", p4, width = 8, height = 5)

# %% AOI-specific summaries
cat("Creating AOI-specific summaries...\n")

aoi_summary <- fixation_clean %>%
  filter(!is.na(aoi_type)) %>%
  group_by(voice_type, aoi_type) %>%
  summarise(
    total_fixations = n(),
    total_duration = sum(CURRENT_FIX_DURATION, na.rm = TRUE),
    mean_duration = mean(CURRENT_FIX_DURATION, na.rm = TRUE),
    .groups = "drop"
  ) %>%
  group_by(voice_type) %>%
  mutate(
    pct_fixations = 100 * total_fixations / sum(total_fixations),
    pct_duration = 100 * total_duration / sum(total_duration)
  )

write.csv(aoi_summary, "results/exploratory/aoi_summary.csv", row.names = FALSE)

# %% Participant-level variability
cat("Calculating participant-level variability...\n")

participant_variability <- fixation_clean %>%
  group_by(participant_id, voice_type, aoi_type) %>%
  summarise(
    mean_fixation_duration = mean(CURRENT_FIX_DURATION, na.rm = TRUE),
    .groups = "drop"
  ) %>%
  group_by(voice_type, aoi_type) %>%
  summarise(
    n_participants = n(),
    mean_across_participants = mean(mean_fixation_duration, na.rm = TRUE),
    sd_across_participants = sd(mean_fixation_duration, na.rm = TRUE),
    .groups = "drop"
  )

write.csv(participant_variability, "results/exploratory/participant_variability.csv", row.names = FALSE)

# %% Print summary
cat("\n=== Exploratory Analysis Summary ===\n")
cat("Summary statistics saved to results/exploratory/\n")
cat("Plots saved to plots/exploratory/\n")
cat("\nExploratory analysis complete!\n")

File: scripts/05_exploratory_analysis.py
# %% Exploratory Data Analysis
# Script: 05_exploratory_analysis.py
# Purpose: Summary statistics and initial data exploration

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

# Set working directory
if not os.path.exists("scripts"):
    os.chdir("..")

# Create output directories
os.makedirs("plots/exploratory", exist_ok=True)
os.makedirs("results/exploratory", exist_ok=True)

# Set matplotlib style
plt.style.use('default')
sns.set_palette("husl")

# Load data with exclusion flags
print("Loading data...")
merged_fixation_data = pd.read_csv("data/processed/merged_fixation_data_with_flags.csv")
merged_interest_area_data = pd.read_csv("data/processed/merged_interest_area_data.csv")
merged_saccade_data = pd.read_csv("data/processed/merged_saccade_data_with_flags.csv")
merged_trial_data = pd.read_csv("data/processed/merged_trial_data_with_flags.csv")

# Filter out excluded data
fixation_clean = merged_fixation_data[~merged_fixation_data['exclude_fixation']].copy()
saccade_clean = merged_saccade_data[~merged_saccade_data['exclude_saccade']].copy()
trial_clean = merged_trial_data[~merged_trial_data['exclude_trial']].copy()

# %% Summary Statistics by Voice Type
print("Calculating summary statistics by voice type...")

# Note: voice_type may need to be extracted from voice_id or other variables
# For now, we'll use voice_id as a proxy if voice_type doesn't exist
if "voice_type" not in trial_clean.columns:
    # Create a placeholder - this should be adjusted based on actual data
    trial_clean['voice_type'] = "unknown"
    fixation_clean['voice_type'] = "unknown"
    saccade_clean['voice_type'] = "unknown"

# Fixation duration summary by voice type and AOI
fixation_summary = fixation_clean.groupby(['voice_type', 'aoi_type']).agg({
    'CURRENT_FIX_DURATION': ['count', 'mean', 'median', 'std', 'min', 'max']
}).reset_index()

fixation_summary.columns = ['voice_type', 'aoi_type', 'n', 'mean_duration', 
                            'median_duration', 'sd_duration', 'min_duration', 'max_duration']

fixation_summary.to_csv("results/exploratory/fixation_summary_by_voice_aoi.csv", index=False)

# Dwell time summary by voice type and AOI
dwell_time_summary = merged_interest_area_data[
    (merged_interest_area_data['IA_LABEL'] != "") & 
    merged_interest_area_data['IA_LABEL'].notna()
].groupby(['voice_id', 'aoi_type']).agg({
    'IA_DWELL_TIME': ['count', 'mean', 'median', 'std']
}).reset_index()

dwell_time_summary.columns = ['voice_id', 'aoi_type', 'n', 'mean_dwell_time', 
                              'median_dwell_time', 'sd_dwell_time']

dwell_time_summary.to_csv("results/exploratory/dwell_time_summary_by_voice_aoi.csv", index=False)

# Saccade summary by voice type
saccade_summary = saccade_clean.groupby('voice_type').agg({
    'CURRENT_SAC_AMPLITUDE': ['count', 'mean'],
    'CURRENT_SAC_DURATION': 'mean',
    'CURRENT_SAC_AVG_VELOCITY': 'mean'
}).reset_index()

saccade_summary.columns = ['voice_type', 'n', 'mean_amplitude', 'mean_duration', 'mean_velocity']

saccade_summary.to_csv("results/exploratory/saccade_summary_by_voice.csv", index=False)

# %% Distribution plots for key dependent variables
print("Creating distribution plots...")

# Fixation duration distribution
plt.figure(figsize=(8, 5))
plt.hist(fixation_clean['CURRENT_FIX_DURATION'], bins=50, color='skyblue', edgecolor='black', alpha=0.6)
plt.title("Distribution of Fixation Duration")
plt.xlabel("Fixation Duration (ms)")
plt.ylabel("Frequency")
plt.tight_layout()
plt.savefig("plots/exploratory/fixation_duration_distribution.png", dpi=300)
plt.close()

# Fixation duration by voice type and AOI
fixation_clean_aoi = fixation_clean[fixation_clean['aoi_type'].notna()].copy()
if len(fixation_clean_aoi) > 0:
    plt.figure(figsize=(10, 6))
    sns.boxplot(data=fixation_clean_aoi, x='voice_type', y='CURRENT_FIX_DURATION', hue='aoi_type')
    plt.title("Fixation Duration by Voice Type and AOI")
    plt.xlabel("Voice Type")
    plt.ylabel("Fixation Duration (ms)")
    plt.legend(title="AOI Type")
    plt.tight_layout()
    plt.savefig("plots/exploratory/fixation_duration_by_voice_aoi.png", dpi=300)
    plt.close()

# Dwell time distribution
dwell_data = merged_interest_area_data[merged_interest_area_data['IA_DWELL_TIME'].notna()].copy()
if len(dwell_data) > 0:
    plt.figure(figsize=(8, 5))
    plt.hist(dwell_data['IA_DWELL_TIME'], bins=50, color='lightgreen', edgecolor='black', alpha=0.6)
    plt.title("Distribution of Dwell Time")
    plt.xlabel("Dwell Time (ms)")
    plt.ylabel("Frequency")
    plt.tight_layout()
    plt.savefig("plots/exploratory/dwell_time_distribution.png", dpi=300)
    plt.close()

# Saccade amplitude distribution
plt.figure(figsize=(8, 5))
plt.hist(saccade_clean['CURRENT_SAC_AMPLITUDE'], bins=50, color='salmon', edgecolor='black', alpha=0.6)
plt.title("Distribution of Saccade Amplitude")
plt.xlabel("Saccade Amplitude (degrees)")
plt.ylabel("Frequency")
plt.tight_layout()
plt.savefig("plots/exploratory/saccade_amplitude_distribution.png", dpi=300)
plt.close()

# %% AOI-specific summaries
print("Creating AOI-specific summaries...")

aoi_summary = fixation_clean[fixation_clean['aoi_type'].notna()].groupby(['voice_type', 'aoi_type']).agg({
    'CURRENT_FIX_DURATION': ['count', 'sum', 'mean']
}).reset_index()

aoi_summary.columns = ['voice_type', 'aoi_type', 'total_fixations', 'total_duration', 'mean_duration']

# Calculate percentages
aoi_summary = aoi_summary.groupby('voice_type').apply(
    lambda x: x.assign(
        pct_fixations=100 * x['total_fixations'] / x['total_fixations'].sum(),
        pct_duration=100 * x['total_duration'] / x['total_duration'].sum()
    )
).reset_index(drop=True)

aoi_summary.to_csv("results/exploratory/aoi_summary.csv", index=False)

# %% Participant-level variability
print("Calculating participant-level variability...")

participant_variability = fixation_clean.groupby(['participant_id', 'voice_type', 'aoi_type']).agg({
    'CURRENT_FIX_DURATION': 'mean'
}).reset_index()

participant_variability.columns = ['participant_id', 'voice_type', 'aoi_type', 'mean_fixation_duration']

participant_variability = participant_variability.groupby(['voice_type', 'aoi_type']).agg({
    'mean_fixation_duration': ['count', 'mean', 'std']
}).reset_index()

participant_variability.columns = ['voice_type', 'aoi_type', 'n_participants', 
                                   'mean_across_participants', 'sd_across_participants']

participant_variability.to_csv("results/exploratory/participant_variability.csv", index=False)

# %% Print summary
print("\n=== Exploratory Analysis Summary ===")
print("Summary statistics saved to results/exploratory/")
print("Plots saved to plots/exploratory/")
print("\nExploratory analysis complete!")

